### PR TITLE
[DOCS] [#11073]: Adding docs for create, refresh, drop matviews

### PR DESCRIPTION
**Pull request by fizaaluthra**
This PR includes documentation for the grammar and semantics of CREATE, REFRESH and DROP statements for materialized views.

### CODE REVIEW

File: docs/content/latest/api/ysql/the-sql-language/statements/ddl_create_matview.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+## Semantics
+
+Create a materialized view named *matview_name*. If `matview_name` already exists in the specified database, an error will be raised unless the `IF NOT EXISTS` clause is used.
+
+### Tablespace
+Used to specify the tablespace for the materialized view.
+
+### Storage parameters
+
+Storage parameters, [as defined by PostgreSQL](https://www.postgresql.org/docs/11/sql-createtable.html#SQL-CREATETABLE-STORAGE-PARAMETERS), are ignored and only present for compatibility with PostgreSQL.
```
Comment (sushantrmishra) [inline docs/content/latest/api/ysql/the-sql-language/statements/ddl_create_matview.md:55]: Should we omit storage parameter from specifying if not applicable ?

File: docs/content/latest/api/ysql/the-sql-language/statements/ddl_create_matview.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+---+---
+ 3 | 4
+(1 row)
+```
+
+## See also
+
+- [`REFRESH MATERIALIZED VIEW`](../refresh_matview)
+- [`DROP MATERIALIZED VIEW`](../ddl_drop_matview)
+- [postgresql-docs-create-matview](https://www.postgresql.org/docs/11/sql-creatematerializedview.html)
```
Comment (sushantrmishra) [inline docs/content/latest/api/ysql/the-sql-language/statements/ddl_create_matview.md:91]: Lets remove this reference to postgres doc. @polarweasel  mentioned that we are trying to minimize the reference to postgres docs unless it is necessary. 
I am not sure if there is value add here by including the reference.

File: docs/content/latest/api/ysql/syntax_resources/the-sql-language/statements/savepoint_rollback.grammar.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-```ebnf
```
Comment (polarweasel) [inline docs/content/latest/api/ysql/syntax_resources/the-sql-language/statements/savepoint_rollback.grammar.md:1]: Curious why you're removing the language from these. Is it not working correctly?

**Review by ymahajan — COMMENTED** [review]
@fizaaluthra - Can we also include Limitations section, and link the corresponding GitHub issues as well.  

- Cannot refresh materialized views inside explicit transactions
- Not supported for YCQL
- MV follows default characteristics of tables like collocation?
- To refresh the MV does it have to manual step or can it be configured to refresh at defined interval?

**Comment by ymahajan** [comment]
@fizaaluthra - We also need to update the 'Views' page in Explore section - https://docs.yugabyte.com/latest/explore/ysql-language-features/advanced-features/views/ 
MV can be a section in this page.

File: docs/content/latest/api/ysql/syntax_resources/the-sql-language/statements/savepoint_rollback.grammar.md (hunk 3 above)
Comment (fizaaluthra) [reply docs/content/latest/api/ysql/syntax_resources/the-sql-language/statements/savepoint_rollback.grammar.md:1]: Hmm, when I regenerate the grammar files using `java -jar rrdiagram.jar content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf   content/latest/api/ysql/syntax_resources/`, these '```ebnf'  lines get removed from the `savepoint_xxx.grammar.md` files. 
Should I just manually check out these files from this PR? or am i running the command incorrectly

Comment (polarweasel) [reply docs/content/latest/api/ysql/syntax_resources/the-sql-language/statements/savepoint_rollback.grammar.md:1]: (I've fixed this in a separate PR, all good. No need for changes here.)

File: docs/content/latest/api/ysql/the-sql-language/statements/ddl_refresh_matview.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+The default mechanism for refreshing a matview is loading the updated contents in a new relation, and then swapping the matview's
+storage with that of the new relation's.
+### WITH DATA
+If `WITH DATA` (default) is specified, the view's query is executed to obtain the new data and the materialized view's contents are updated.
+
+### WITH NO DATA
+If `WITH NO DATA` is specified, the old contents of the materialized view are discarded and the materialized view is left in an unscannable state.
+
+### CONCURRENTLY
+This option differs in the mechanism used to perform the refresh. When `CONCURRENTLY` is specified, a diff between the old data and the new data is computed, and the required updates are performed accordingly.
```
Comment (sushantrmishra) [inline docs/content/latest/api/ysql/the-sql-language/statements/ddl_refresh_matview.md:59]: Ideally we should also  specify or connect the dots to specify  the scenario when concurrently is performs better  and when it does not.
From PG docs  -->
This option may be faster in cases where a small number of rows are affected. Without this option a refresh which affects a lot of rows will tend to use fewer resources and complete more quickly

**Review by sushantrmishra — APPROVED** [review]

**Comment by polarweasel** [comment]
@fizaaluthra can you please regenerate the diagrams? There's a new version of the RRDiagram tool that changes the markdown output slightly.

**Pull request merged**